src/plugins/auto_batch/src/compiled_model.cpp

```diff
@@ -80,10 +80,8 @@ CompiledModel::GetWorkerInferRequest() const {
         workerRequestPtr->_thread = std::thread([workerRequestPtr, this] {
             while (1) {
                 std::cv_status status;
-                {
-                    std::unique_lock<std::mutex> lock(workerRequestPtr->_mutex);
-                    status = workerRequestPtr->_cond.wait_for(lock, std::chrono::milliseconds(m_time_out));
-                }
+                std::unique_lock<std::mutex> lock(workerRequestPtr->_mutex);
+                status = workerRequestPtr->_cond.wait_for(lock, std::chrono::milliseconds(m_time_out));
```

Comment:
I suppose mutex should be released after wait_for, should not it?

Reply:
brief look seems we don't need to move this out of the code block, what  is the original complain of this issue?

Reply:
@songbell  @ilya-lavrenov 
the static analyzer is complaining about is wait_for() not being in the loop.
![image](https://github.com/user-attachments/assets/a9a927f4-01c0-46be-84c7-cb734a6c7e91)

Reply:
https://juniorfans.gitbooks.io/blog/content/3-condvar-spurious-wakeups.html